### PR TITLE
ui-testing

### PR DESCRIPTION
**Pull request by naumvd95**
proofs = http://jenkins-tpi.bud.mirantis.net:8080/view/ui/job/10.0.ui.test_bonds/17/console

### CODE REVIEW

File: servers/partner-ci/builders/common.prepare.sh [hunk 1]
```diff
@@ -20,15 +20,16 @@ fi
...
-if [[ $ISO_FILE == *"custom"* ]]; then
+else
   export FUEL_RELEASE=90
 fi
 
-if [[ "${UPDATE_MASTER}" == "true" ]] && [[ ${FUEL_RELEASE} != *"80"* ]]; then
+if [[ "${UPDATE_MASTER}" == "true" ]] && [[ ${FUEL_RELEASE} != *"80"* ]] && [[ ${FUEL_RELEASE} != *"100"* ]]; then
   [ ${SNAPSHOTS_ID} ] && export SNAPSHOTS_ID=${SNAPSHOTS_ID} || export SNAPSHOTS_ID=${CUSTOM_VERSION:10}
+elif [[ ${FUEL_RELEASE} == *"100"* ]]; then
+  export SNAPSHOTS_ID="from_iso"
```
Comment (naumvd95) [inline servers/partner-ci/builders/common.prepare.sh]: pay attention!!
there are some troubles with it
need to refactor testing-code
**2017-02-13 14:29:55 - INFO environment.py:237 -- Try to find environment '10.0.ui.released''**

**Review by rkhozinov — CHANGES_REQUESTED** [review]
Separate download job to another pull-request

File: servers/partner-ci/builders/common.prepare.sh [hunk 2]
```diff
@@ -20,15 +20,16 @@ fi
 [ -z $ISO_FILE ] && (echo "ISO_FILE variable is empty"; exit 1)
 
 
-if [[ $ISO_FILE == *"Mirantis"* ]]; then
+if [[ $ISO_FILE == *"Mirantis"* ]] || [[ "${FUEL_RELEASE_VERSION}" == "10" ]] ; then
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: join conditions

File: servers/partner-ci/builders/common.prepare.sh (hunk 1 above)
Comment (rkhozinov) [reply servers/partner-ci/builders/common.prepare.sh]: **10.0.ui.released** --> **10.0.ui.<iso_number>**

File: servers/partner-ci/common.yaml [hunk 3]
```diff
@@ -194,6 +194,10 @@
           name: SNAPSHOTS_ID
           default: '{snapshots_id}'
           description: 'snapshots number, "lastSuccessfulBuild" or "released"'
+      - string:
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml]: add to common props

File: servers/partner-ci/common.yaml [hunk 4]
```diff
@@ -630,7 +634,7 @@
           properties-file: properties_file.txt
       - shell: |
           #!/bin/bash -xeu
-          UBUNTU_MIRROR_URL="${MIRROR_HOST:?}${UBUNTU_MIRROR_ID:?}/"
+          UBUNTU_MIRROR_URL="${MIRROR_HOST:?}/pkgs/snapshots/${UBUNTU_MIRROR_ID:?}/"
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml]: add to the common props the next property: 
**MIRROR_HOST=http://mirror.seed-cz1.fuel-infra.org/pkgs/snapshots/**

File: servers/partner-ci/common.yaml [hunk 5]
```diff
@@ -641,7 +645,7 @@
               UBUNTU_REPOS="$UBUNTU_REPOS|$UBUNTU_PROPOSED"
           fi
 
-          echo "MIRROR_UBUNTU="${UBUNTU_REPOS:?}" >> specific.properties
+          echo MIRROR_UBUNTU="${UBUNTU_REPOS:?}" >> specific.properties
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml]: "MIRROR_UBUNTU=${UBUNTU_REPOS:?}"

Comment (rkhozinov) [reply servers/partner-ci/common.yaml]: java properties: 

MIRROR=asdfadsf

bash properties
MIRROR="asdf"

Comment (rkhozinov) [reply servers/partner-ci/common.yaml]: `echo "MIRROR_UBUNTU=${UBUNTU_REPOS:?}"`

File: servers/partner-ci/builders/ui.test.sh [hunk 6]
```diff
@@ -31,7 +31,7 @@ sh -ex "utils/jenkins/system_tests.sh"  \
    -w "$(pwd)"                          \
    -t test                              \
    -o --group="${FUEL_QA_TEST_GROUP:?}" \
-   -i ${ISO_FILE:?}
+   -i ${ISO_STORAGE}/${ISO_FILE:?}
```
Comment (rkhozinov) [inline servers/partner-ci/builders/ui.test.sh]: `-i ${ISO_STORAGE:?}/${ISO_FILE:?}`

File: servers/partner-ci/builders/common.prepare.sh [hunk 7]
```diff
@@ -45,21 +46,19 @@ if [[ $ISO_FILE == *"Mirantis"* ]]; then
...
-if [[ $ISO_FILE == *"custom"* ]]; then
-  export FUEL_RELEASE=90
-fi
-
 export REQUIRED_FREE_SPACE=200
 
 export ENV_NAME="${ENV_PREFIX}.${SNAPSHOTS_ID}"
 export VENV_PATH=${VENV_PATH:-"$HOME/$FUEL_RELEASE-venv"}
+
+[[ -d $VENV_PATH ]] || virtualenv --clear "${VENV_PATH}"
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: `[[ -d $VENV_PATH ]] `--> `[ -d $VENV_PATH ] `OR `test -d $VENV_PATH`

**Review by rkhozinov — CHANGES_REQUESTED** [review]

File: servers/partner-ci/builders/common.prepare.sh (hunk 2 above)
Comment (naumvd95) [reply servers/partner-ci/builders/common.prepare.sh]: there is different conditions
need to refactor this cases
i will make the issue

Comment (naumvd95) [reply servers/partner-ci/builders/common.prepare.sh]: https://github.com/rkhozinov/jenkins-jobs/issues/175

File: servers/partner-ci/builders/common.prepare.sh (hunk 1 above)
Comment (naumvd95) [reply servers/partner-ci/builders/common.prepare.sh]: done

File: servers/partner-ci/builders/common.prepare.sh (hunk 7 above)
Comment (naumvd95) [reply servers/partner-ci/builders/common.prepare.sh]: done

File: servers/partner-ci/builders/ui.test.sh (hunk 6 above)
Comment (naumvd95) [reply servers/partner-ci/builders/ui.test.sh]: done

File: servers/partner-ci/common.yaml (hunk 3 above)
Comment (naumvd95) [reply servers/partner-ci/common.yaml]: done

File: servers/partner-ci/common.yaml (hunk 4 above)
Comment (naumvd95) [reply servers/partner-ci/common.yaml]: done

File: servers/partner-ci/common.yaml (hunk 5 above)
Comment (naumvd95) [reply servers/partner-ci/common.yaml]: done

File: servers/partner-ci/builders/build-job-download-iso.sh [hunk 8]
```diff
@@ -8,20 +8,20 @@ FUEL_RELEASE=$(echo $VERSION | tr -d . )
 echo fuel release $FUEL_RELEASE, version $VERSION
 echo "Description string: $NODE_NAME $VERSION"
 
+ISO_STORAGE="/storage/downloads"
 aria2c --seed-time 0 "${MAGNET_LINK}" -d "${WORKSPACE}"
```
Comment (rkhozinov) [inline servers/partner-ci/builders/build-job-download-iso.sh:12]: why you don't download iso directly to `/storage/downloads`?

File: servers/partner-ci/builders/common.prepare.sh [hunk 9]
```diff
@@ -45,21 +48,19 @@ if [[ $ISO_FILE == *"Mirantis"* ]]; then
...
 export VENV_PATH=${VENV_PATH:-"$HOME/$FUEL_RELEASE-venv"}
+
+test -d $VENV_PATH || virtualenv --clear "${VENV_PATH}"
 ## For plugins we should get a valid version of requrements of python-venv
 ## This requirements could be got from the github repo
 ## but for each branch of a plugin we should map specific branch of the fuel-qa repo
 ## the fuel-qa branch is determined by a fuel-iso name.
 
 case "${FUEL_RELEASE}" in
-  *10* ) export REQS_BRANCH="stable/newton" ;;
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: join with line 30

**Review by rkhozinov — CHANGES_REQUESTED** [review]

File: servers/partner-ci/builders/common.prepare.sh (hunk 2 above)
Comment (rkhozinov) [reply servers/partner-ci/builders/common.prepare.sh]: `FUEL_RELEASE_VERSION` - where do we initialize the var?

**Review by rkhozinov — CHANGES_REQUESTED** [review]

File: servers/partner-ci/builders/ui.test.sh [hunk 10]
```diff
@@ -6,7 +6,7 @@ fuel_release=$(echo $ISO_FILE | cut -d- -f2 | tr -d '.iso')
 export FUEL_RELEASE=${fuel_release:?}
 
 if [ "${SNAPSHOTS_ID}" != "released" ]; then
-  if [[ "${UPDATE_MASTER}" == "true" ]] && [[ ${FUEL_RELEASE} != *"80"* ]]; then
+  if [[ "${UPDATE_MASTER}" == "true" ]] && [[ ${FUEL_RELEASE} != *"80"* ]] && [[ ${FUEL_RELEASE} != *"100"* ]]; then
```
Comment (rkhozinov) [inline servers/partner-ci/builders/ui.test.sh]: try to use switch or regex groups

File: servers/partner-ci/common.yaml [hunk 11]
```diff
@@ -119,16 +115,17 @@
             FORCE_DISABLE_UPDATES=false;
             FUEL_STATS_ENABLED=false;
             GIT_CURL_VERBOSE=false;
+            ISO_STORAGE=/storage/downloads;
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml:118]: ok

File: servers/partner-ci/common.yaml [hunk 12]
```diff
@@ -119,16 +115,17 @@
...
             MAKE_SNAPSHOT=true;
             NODES_COUNT=10;
             OPENSTACK_RELEASE=Ubuntu;
             SLAVE_NODE_CPU=4;
             SLAVE_NODE_MEMORY=4096;
             SNAPSHOT_ARTIFACTS_FILE=snapshots.params;
             SNAPSHOT_OUTPUT_FILE=extra_repos.sh;
+            VENV_PATH=$HOME/90-venv;
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml]: 90-venv is used for fuel90
100-venv is used for fuel100

VENV_PATH wasn't a constant

File: servers/partner-ci/common.yaml [hunk 13]
```diff
@@ -641,7 +639,7 @@
               UBUNTU_REPOS="$UBUNTU_REPOS|$UBUNTU_PROPOSED"
           fi
 
-          echo "MIRROR_UBUNTU="${UBUNTU_REPOS:?}" >> specific.properties
+          echo "MIRROR_UBUNTU=${UBUNTU_REPOS:?}" >> specific.properties
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml]: ok

File: servers/partner-ci/common.yaml [hunk 14]
```diff
@@ -210,10 +207,11 @@
             FUEL_RELEASE_VERSION={fuel_release_checker};
             FUEL_STATS_ENABLED=false;
             GIT_CURL_VERBOSE=false;
+            ISO_STORAGE=/storage/downloads;
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml:206]: ok

File: servers/partner-ci/common.yaml [hunk 15]
```diff
@@ -210,10 +207,11 @@
...
             ISO_JOB_NAME=download_iso;
             KVM_USE={kvm_use};
             MAKE_SNAPSHOT=true;
-            MIRROR_HOST=mirror.seed-cz1.fuel-infra.org;
+            MIRROR_HOST=http://mirror.seed-cz1.fuel-infra.org/pkgs/snapshots/;
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml:210]: ok

File: servers/partner-ci/builders/common.prepare.sh [hunk 16]
```diff
@@ -20,22 +20,25 @@ fi
...
+    *100* ) export SNAPSHOTS_ID=$(echo $ISO | cut -d- -f3 | tr -d '.iso') ;;
+    *90* ) export SNAPSHOTS_ID=${${SNAPSHOTS_ID}:=${CUSTOM_VERSION:10}} ;;
+    *80* ) echo "incompatible version for snapshots_id" ;;
+    *70* ) echo "incompatible version for snapshots_id" ;;
+    *61* ) echo "incompatible version for snapshots_id" ;;
+     *   ) export SNAPSHOTS_ID="released"
+  esac
 fi
 
 [ -z "${SNAPSHOTS_ID}" ] && { echo SNAPSHOTS_ID is empty; exit 1; }
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: remove the check `[ -z "${SNAPSHOTS_ID}" ] && { echo SNAPSHOTS_ID is empty; exit 1; }`

File: servers/partner-ci/builders/common.prepare.sh [hunk 17]
```diff
@@ -20,22 +20,25 @@ fi
 [ -z $ISO_FILE ] && (echo "ISO_FILE variable is empty"; exit 1)
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: remove `[ -z $ISO_FILE ] && (echo "ISO_FILE variable is empty"; exit 1)`
and
change 
`if [[ ${ISO_FILE:?} == *"Mirantis"* ]] || [[ "${FUEL_RELEASE_VERSION}" == "10" ]] `

**Review by rkhozinov — CHANGES_REQUESTED** [review]

File: servers/partner-ci/builders/build-job-download-iso.sh (hunk 8 above)
Comment (naumvd95) [reply servers/partner-ci/builders/build-job-download-iso.sh:12]: we already talk about that
before moving we need to define ISO_FILE=$(ls *.iso)

File: servers/partner-ci/builders/common.prepare.sh (hunk 2 above)
Comment (naumvd95) [reply servers/partner-ci/builders/common.prepare.sh]: common.yaml:            FUEL_RELEASE_VERSION={fuel_release_checker};

File: servers/partner-ci/builders/common.prepare.sh (hunk 17 above)
Comment (naumvd95) [reply servers/partner-ci/builders/common.prepare.sh]: done

File: servers/partner-ci/builders/common.prepare.sh (hunk 9 above)
Comment (naumvd95) [reply servers/partner-ci/builders/common.prepare.sh]: done

File: servers/partner-ci/builders/ui.test.sh (hunk 10 above)
Comment (naumvd95) [reply servers/partner-ci/builders/ui.test.sh]: done

File: servers/partner-ci/common.yaml (hunk 12 above)
Comment (naumvd95) [reply servers/partner-ci/common.yaml]: this is common variable, seems this is right
in spec-props i'll define VENV_PATH=$HOME/100-venv;

File: servers/partner-ci/builders/build-job-download-iso.sh (hunk 8 above)
Comment (rkhozinov) [reply servers/partner-ci/builders/build-job-download-iso.sh:12]: aria2c provides information about downloaded files

File: servers/partner-ci/builders/common.prepare.sh [hunk 18]
```diff
@@ -165,38 +105,15 @@ function dospy_list {
...
-wait_ws() {
-  while [ $(pgrep vmrun | wc -l) -ne 0 ] ; do sleep 5; done
-}
-
 if [[ "${RECREATE_VENV}" == "true" ]]; then
   recreate_venv
   get_venv_requirements
-  [ -d $VENV_PATH ] && prepare_venv || { echo "$VENV_PATH doesn't exist $VENV_PATH will be recreated"; recreate_venv; }
+  [ -d $VENV_PATH ] && prepare_venv || \
+    { echo "$VENV_PATH doesn't exist $VENV_PATH will be recreated"; recreate_venv; }
```
Comment (rkhozinov) [inline servers/partner-ci/builders/common.prepare.sh]: the statement's logic confuses

**Review by rkhozinov — CHANGES_REQUESTED** [review]

**Review by rkhozinov — APPROVED** [review]

File: servers/partner-ci/9.0/contrail.yaml [hunk 19]
```diff
@@ -392,11 +392,12 @@
...
-      - get-refs
-      - get-updates
-      - get-baremetal-props
-      - get-props
-      - get-snapshots
+      - 10-get-refs
+      - 20-get-props
+      - 21-get-baremetal-props
+      - 30-get-additional-updates
+      - 40-get-snapshots
```
Comment (rkhozinov) [inline servers/partner-ci/9.0/contrail.yaml]: join additional updates with get-snapshots

**Review by rkhozinov — CHANGES_REQUESTED** [review]

File: servers/partner-ci/common.yaml [hunk 20]
```diff
@@ -589,6 +566,23 @@
           steps:
             - shell: |
                 #!/bin/bash -xeu
+                if [[ "${UPDATE_MASTER}" == "true" ]]; then
+                cat << SPECIFIC_PROPERTIES > specific.properties
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml]: indent

File: servers/partner-ci/common.yaml [hunk 21]
```diff
@@ -598,16 +587,22 @@
                 echo "snapshot_id_number=${snapshots_id}"
                 python "${CONV_SNAP_PATH:?}"
 
+                . $SNAPSHOT_OUTPUT_FILE
+                echo "EXTRA_RPM_REPOS=${EXTRA_RPM_REPOS}" >> get-snapshots.properties
```
Comment (rkhozinov) [inline servers/partner-ci/common.yaml]: optimize multiple ehos with cat

**Review by rkhozinov — CHANGES_REQUESTED** [review]

**Review by rkhozinov — APPROVED** [review]

**Pull request merged**